globus_automate_client/cli/auth.py

```diff
@@ -296,8 +296,6 @@ def refresh_handler(
                     expires_at=token_set.expiration_time,
                     on_refresh=refresh_handler,
                 )
```

Comment:
I think we were doing this to force the authorizer to refresh its token if it was expired. It looks like this got renamed and is getting called implicitly in `get_authorization_header`. So it sounds like we don't actually need to do this.

@jimPruyne can you remember a specific reason we were doing this ourselves? If not, I'm happy removing this and trusting the RefreshingAuthorizer to do its thing.

Reply:
This is correct; `.get_authorization_header` should handle this.

Reply:
In fact, with the 3.0 additions, if you have an authorizer attached to your `BaseClient `implementation I believe none of that is even necessary; `RequestsTransport.request` is handling it all.

Reply:
Thanks for the review, Rudy! To confirm: you're saying that removing this call is okay? Or is there something I need to address here?

Reply:
The original purpose was, I believe, as @urimandujano says. If that is a changed behavior in SDK3, as @rudyardrichter says, I think I agree with you @kurtmckee , but I would also need clarification.

Reply:
I may be reading this incorrectly, but it looks like the expiration time is checked here, in the `ensure_valid_token()` method. This is indeed called from within `RenewingAuthorizer.get_authorization_header()`.

https://github.com/globus/globus-sdk-python/blob/main/src/globus_sdk/authorizers/renewing.py#L138

It's possible to call `ensure_valid_token()` explicitly. Does that need to happen here?

Reply:
Kurt has it. I can link more, specific pieces that tie into this. (The links are to my fork, sorry; just easier to generate from my editor.)

- `BaseClient.request` (used for `BaseClient.get` et al) [uses this intermediate `RequestsTransport.request`](https://github.com/rudyardrichter/globus-sdk-python/blob/a346def64ce6b1490ea38aa62b92f884f01552bd/src/globus_sdk/client.py#L282).
- `RequestsTransport.request` [calls `RequestsTransport._set_authz_header`](https://github.com/rudyardrichter/globus-sdk-python/blob/a346def64ce6b1490ea38aa62b92f884f01552bd/src/globus_sdk/transport/requests.py#L231)
- `RequestsTransport._set_authz_header` [calls `GlobusAuthorizer.get_authorization_header`](https://github.com/rudyardrichter/globus-sdk-python/blob/a346def64ce6b1490ea38aa62b92f884f01552bd/src/globus_sdk/transport/requests.py#L174)
- ...which finally, **if you are using a `RenewingAuthorizer` and not overriding `get_authorization_header`,** makes sure the token works.